File: app/src/main/java/com/gnardini/tvshowcontroller/mouse_movement/MouseActivity.java
package com.gnardini.tvshowcontroller.mouse_movement;

import android.app.Activity;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;

import com.gnardini.tvshowcontroller.R;

import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.OnTouch;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MouseActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.mouse_activity);
        ButterKnife.bind(this);
    }

    @OnTouch(R.id.mouse_input)
    void onMouseMoved(View view, MotionEvent motionEvent) {

    }

    @OnClick(R.id.left_click)
    void onLeftClick() {

    }

    @OnClick(R.id.right_click)
    void onRightClick() {

    }

    private Callback<String> emptyCallback = new Callback<String>() {
        @Override
        public void onResponse(Call<String> call, Response<String> response) {
        }

        @Override
        public void onFailure(Call<String> call, Throwable t) {
        }
    };


}

File: app/src/main/java/com/gnardini/tvshowcontroller/mouse_movement/MouseActivity.kt
package com.gnardini.tvshowcontroller.mouse_movement

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MotionEvent
import com.gnardini.tvshowcontroller.R
import com.gnardini.tvshowcontroller.TvShowControllerApp
import com.gnardini.tvshowcontroller.extensions.setClickListener
import com.gnardini.tvshowcontroller.extensions.setOnTouchListener

class MouseActivity : AppCompatActivity() {

    private val commonInjector = TvShowControllerApp.commonInjector

    private val controllerInteractor = commonInjector.controllerInteractors
    private val mouseMovementTracker = commonInjector.mouseMovementTracker

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.mouse_activity)

        setOnTouchListener(R.id.mouse_input, this::onMouseMoved)
        setClickListener(R.id.left_click, controllerInteractor::leftClick)
        setClickListener(R.id.right_click, controllerInteractor::rightClick)
    }

    private fun onMouseMoved(motionEvent: MotionEvent) {
        val x = motionEvent.x
        val y = motionEvent.y
        when (motionEvent.action) {
            MotionEvent.ACTION_DOWN-> mouseMovementTracker.startMovement(x, y)
            MotionEvent.ACTION_UP -> mouseMovementTracker.endMovement(x, y)
            MotionEvent.ACTION_MOVE -> mouseMovementTracker.onMovement(x, y)
        }
    }

}
